packages/terra-folder-tree/src/FolderTree.jsx

```diff
@@ -128,6 +139,20 @@ const FolderTree = ({
     listItems[0].tabIndex = 0;
   }, []);
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => {
+    const listItems = listNode.current.querySelectorAll('[data-item-show-focus=true]');
+    const visibleListItems = getVisibleListItems(listItems);
+
+    if (visibleListItems.length === children.length) {
+      setAriaLiveMessage(intl.formatMessage({ id: 'Terra.folder-tree.button.collapse-all-announcement' }));
+    }
+
+    if (visibleListItems.length === listItems.length) {
+      setAriaLiveMessage(intl.formatMessage({ id: 'Terra.folder-tree.button.expand-all-announcement' }));
+    }
+  });
```

Comment:
It looks like the dependency array is missing here.  This did not trigger a linter error?

Reply:
It did trigger a linter error, which is disabled above. The reason it's being disabled is to ensure the useEffect runs after `onExpandAll`/`onCollapseAll` trigger a re-render. Including the required dependency array prevents this.

Obviously the side effect is that this hook has to run on every render, but I haven't been able to find a way around that so far. I suppose a workaround could be to include a dummy state variable that updates when `onExpandAll`/`onCollapseAll` are called, thoughts?

Reply:
On second thought, I'm not sure a dummy state variable will work either, since we only want this to trigger when the items have actually been expanded, not just if the button was clicked.

Reply:
Still need to fix tests, but updated to add back the dependency array and trigger the useEffect only on `visibleListItems` change:

[c093e25](https://github.com/cerner/terra-framework/pull/2005/commits/c093e25686488d26a43dd4b2d16f53a6f0d1f7a2)

Thoughts?

Reply:
It's a nice solution for expand all/collapse all, but it looks like it is not working with the tree items being opened one by, or am I missing something?

Reply:
Based on the Jira's AC the announcement is for when the expand all/collapse all buttons are pressed, not just when the whole tree is expanded item by item. Checked with @eawww who confirmed that having the announcement on just the buttons (as long as the items are actually expanded/collapsed) is acceptable.